packages/x-charts/src/ChartsVoronoiHandler/ChartsVoronoiHandler.tsx

```diff
@@ -107,10 +107,10 @@ function ChartsVoronoiHandler(props: ChartsVoronoiHandlerProps) {
   }, [defaultXAxisId, defaultYAxisId, series, seriesOrder, xAxis, yAxis, drawingArea]);
 
   React.useEffect(() => {
-    const element = svgRef.current;
-    if (element === null) {
+    if (svgRef.current === null) {
       return undefined;
     }
+    const element = svgRef.current;
```

Comment:
How could it be? Between the if and the assignment there is no thing, So I don't see how the main JS thread could be interupted

Reply:
That's why I said "technically" 😅 

The ref is a mutable ref, so we can technically change it at any time, if there are async processes that set/remove it, then it could change.

Reply:
> if there are async processes that set/remove it, then it could change.

Not fully sure, but for me JS is mono-thread, which means the current function is executed up to its end, or an `await`. Only after finishing this execution, the event loop will take other tasks to work on. That's why we sometime use the `setTimeout(..., 0)` By running a code in async way, we are sure to be executed after the end of React main script